src/core/inspect/bimap.rs

```diff
@@ -0,0 +1,52 @@
+use std::borrow::Borrow;
+use std::collections::{btree_map::Range, BTreeMap};
+use std::ops::RangeBounds;
+use std::sync::Arc;
```

Comment:
nit: not really sure if it's an extended practice, but we  tend to group these kind of lines together as in:
```rust
use std::{
   borrow::Borrow,
   collections::{btree_map::Range, BTreeMap};
   ops::RangeBounds;
   sync::Arc,
};
```

Reply:
Sure thing!